test/e2e/run/e2e.go

```diff
@@ -0,0 +1,623 @@
+package run
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/go-logr/logr"
+	ginkgoTypes "github.com/onsi/ginkgo/v2/types"
+	"gopkg.in/yaml.v2"
+
+	"github.com/aws/eks-hybrid/test/e2e"
+	"github.com/aws/eks-hybrid/test/e2e/cleanup"
+	"github.com/aws/eks-hybrid/test/e2e/cluster"
+	"github.com/aws/eks-hybrid/test/e2e/constants"
+	"github.com/aws/eks-hybrid/test/e2e/suite"
+)
+
+const (
+	e2eConfigFolder      = "configs"
+	e2eConfigFile        = "e2e-param.yaml"
+	e2eTestResourcesFile = "e2e-test-resources.yaml"
+	e2eReportsFolder     = "reports"
+	e2eReportsFile       = "junit-nodeadm.xml"
+	e2eReportsFileJSON   = "junit-nodeadm.json"
+	serialOutputLogFile  = "serial-output.log"
+	testCleanupLogFile   = "cleanup-output.log"
+	testGinkgoOutputLog  = "ginkgo-output.log"
+	testSetupLogFile     = "setup-output.log"
+)
+
+type E2E struct {
```

Comment:
can we split the functionality of E2E into multiple pieces? This file a bit difficult to review and I think it's because everything is one object, so it requires to keep the whole thing in your head. I think if we split into multiple independent parts it would help read and modify this code.

We can still have one single `E2E.Run` that makes sense. But we could have a test runner, a test result processor, an artifact uploader... Not feeling strong about that separation in particular, just giving examples for how it could be split.